src/main/java/backend/github/ReplaceIssueMilestoneTask.java

```diff
@@ -0,0 +1,38 @@
+package backend.github;
+
+import backend.interfaces.Repo;
+import backend.interfaces.TaskRunner;
+import org.eclipse.egit.github.core.Issue;
+
+import java.io.IOException;
+
+public class ReplaceIssueMilestoneTask extends GitHubRepoTask<Boolean> {
+    private final String repoId;
+    private final int issueId;
+    private final String issueTitle;
+    private final Integer issueMilestone;
+
+    public ReplaceIssueMilestoneTask(TaskRunner taskRunner, Repo repo, String repoId, int issueId, String issueTitle,
+                                     Integer issueMilestone) {
+        super(taskRunner, repo);
+        this.repoId = repoId;
+        this.issueId = issueId;
+        this.issueMilestone = issueMilestone;
+        this.issueTitle = issueTitle;
+    }
+
+    @Override
+    public void run() {
+        try {
+            Issue result = repo.setMilestone(repoId, issueId, issueTitle, issueMilestone);
+
+            if (result.getMilestone() == null) {
+                response.complete(issueMilestone == null);
```

Comment:
This part here is checking the _resulting issue_ returned from GitHub, whether it was correctly/successfully changed.

I think the problem here is the way the GitHub API handles milestones, it either has to be:
- issue.getMilestone() is null if no milestone is assigned to the issue
- issue.getMilestone().getNumber() is the milestone's id if a milestone is assigned to the issue

So the checking of null is because we have to separately handle the case when no milestone is assigned, and we cannot just invoke issue.getMilestone().getNumber() since getMilestone may return null.